### ISSUE TITLE
Category with multiple labels produces a header per label

### ISSUE THREAD

**Issue by JamesMGreene** [release-drafter/release-drafter#293]
Using the latest (`v5.2.0`) as a GitHub Action in my repo's workflow, I am seeing adverse behavior from #219 

My `release-drafter.yml` configuration file:

```yaml
name-template: '$NEXT_PATCH_VERSION'
tag-template: '$NEXT_PATCH_VERSION'
categories:
  - title: '🚀 Features'
    labels:
      - 'enhancement'
  - title: '🐛 Bug Fixes'
    labels:
      - 'bug'
  - title: '🛠️ Maintenance'
    labels:
      - 'maintenance'
      - 'dependencies'
template: |
  ## What's Changed

  $CHANGES
``` 

Notice that the category with the title "Maintenance" has 2 labels under it: "maintenance" and "dependencies".

When I have a Draft Release open that features merged PRs from _each_ of those labels, the Draft Release contents end up with a duplicate header:

![image](https://user-images.githubusercontent.com/417751/63625782-45fe4900-c5c6-11e9-876f-d900d650c5b0.png)

**Comment by casz** [comment]
My guesss is @JamesMGreene you have two labels of `maintenance` with very similar spelling just different casing.

**Comment by casz** [comment]
Please link to the line so that someone else can help

**Comment by JamesMGreene** [comment]
For my config, the [`getCategoriesConfig` method](https://github.com/toolmantim/release-drafter/blob/c918a5a4e92371b511310d3f5616d3ec41671bec/lib/releases.js#L80) returns the following:

**Result `[0]`** (`categoriesConfig`):

```json
{
  "enhancement": {
    "title": "🚀 Features",
    "labels": ["enhancement"]
  },
  "bug": {
    "title": "🐛 Bug Fixes",
    "labels": ["bug"]
  },
  "maintenance": {
    "title": "🛠️ Maintenance",
    "labels": ["maintenance", "dependencies"]
  },
  "dependencies": {
    "title": "🛠️ Maintenance",
    "labels": ["maintenance", "dependencies"]
  }
}
```

**Result `[1]`** (`orderedCategories`):

```json
["UNCATEGORIZED", "enhancement", "bug", "maintenance", "dependencies"]
```

---

The first input to the [`categorizePullRequests` method](https://github.com/toolmantim/release-drafter/blob/c918a5a4e92371b511310d3f5616d3ec41671bec/lib/releases.js#L97) has input like the following:

**`pullRequests`:**

```json
[
  {
    "title": "Title A",
    "number": 1,
    "author": {
      "login": "JamesMGreene"
    },
    "labels": {
      "nodes": [{ "name": "maintenance"}]
    }
  },
  {
    "title": "Title B",
    "number": 2,
    "author": {
      "login": "dependabot"
    },
    "labels": {
      "nodes": [{ "name": "dependencies"}]
    }
  }
]
```

And results in the following `categorizedPullRequests`:

```json
{
  "UNCATEGORIZED": [],
  "maintenance": [
    {
      "title": "Title A",
      "number": 1,
      "author": {
        "login": "JamesMGreene"
      },
      "labels": {
        "nodes": [
          {
            "name": "maintenance"
          }
        ]
      }
    }
  ],
  "dependencies": [
    {
      "title": "Title B",
      "number": 2,
      "author": {
        "login": "dependabot"
      },
      "labels": {
        "nodes": [
          {
            "name": "dependencies"
          }
        ]
      }
    }
  ]
}
```

---

So the following 2 places are points of interest for fixing this bug because, despite their suggestive names, the keys in both objects are _actually_ oriented by labels, _NOT_ by the configured categories (i.e. by category titles, I suppose):
 - `categoriesConfig` result
 - `categorizedPullRequests` result

**Comment by JamesMGreene** [comment]
Finally, the rendering loops through the [`orderedCategories` array](https://github.com/toolmantim/release-drafter/blob/c918a5a4e92371b511310d3f5616d3ec41671bec/lib/releases.js#L135-L162), which contains values that are _actually_ label names rather than category titles, so for each of the multiple labels associated with a single category, it generates the associated Markdown category title once for each label rather than once for the category.

**Comment by ssbarnea** [comment]
I encountered the same bug on https://github.com/ansible/molecule/releases -- is there a way to address it?

**Comment by casz** [comment]
Don't have over lapping title for labels.

**Comment by erikmueller** [comment]
But isn’t it the other way round (we have overlapping labels for one title)? 
And this was a quite recent addition IIRC. Are we using it wrong?

**Comment by ssbarnea** [comment]
I am not sure how it was implemented but I think that a reasonable expectation for someone looking at the configuration example is that the tool will list changes in the first category that has a matching label. 

* A change should never be listed twice.
* Categories should be listed as defined in template
* An empty category would not be listed

**Comment by LucienShui** [comment]
I have the same problem, my `release-drafter.yml` looks like below:

```yaml
name-template: 'release-v$NEXT_PATCH_VERSION'
tag-template: 'release-v$NEXT_PATCH_VERSION'
categories:
  - title: 'Features'
    labels:
      - 'feature'
      - 'enhancement'
  - title: 'Bug Fixes'
    labels:
      - 'fix'
      - 'bugfix'
      - 'bug'
  - title: 'Maintenance'
    labels:
      - 'chore'
      - 'documentation'
change-template: '- $TITLE (#$NUMBER) @$AUTHOR'
template: |
  # Changes

  $CHANGES
```

When I create two pull requests with label bug and fix, it draft two same titles, one contains fix only, another contains bug only.
Here is the [link](https://github.com/PasteUs/HelloCI/releases/tag/release-v1.0.5)

**Comment by niveathika** [comment]
Faced the same issue when testing this out

**Comment by TimonVS** [comment]
Hey, sorry for taking so long to respond. I've been able to reproduce this issue and I'll tackle it very soon.

**Comment by casz** [comment]
Tackled in #412

**Comment by TimonVS** [comment]
Thanks to @casz this has been fixed and released in [`v5.7.0`](https://github.com/release-drafter/release-drafter/releases/tag/v5.7.0)!
